polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1546 @@
+                if self.dist(firstpoint,midpoint) < self.merge_separation:
+                    firstpoint = midpoint
+                    ii += 1
+                    continue
+
+                # Merging the mid and last point if separation close, determine new edge for adjacency
+                if self.dist(midpoint,lastpoint) < self.merge_separation:
+                    start_cell  = ap.start
+                    end_cell    = app.end
+
+                    common_cell = np.where(np.array(start_cell['neighbourIndex']) == end_cell['id'])[0]
+                    if len(common_cell) == 1:
+                        _merge_case = start_cell['case'][np.where(np.array(start_cell['neighbourIndex']) == end_cell['id'])[0][0]]
+                        new_edge = find_edge(start_cell,end_cell,_merge_case)
+                        self.remove(ii) #Removing ap
+                        self.remove(ii) #Removing app
+                        self.add(ii,[new_edge])
+                        path_length -= 1
+                        converged = False
+                        continue
+
+
+                # Relationship is a diagonal case
+                if self.diagonal_case(ap.case):
+                    add_indicies,add_cases = self.diagonal_select_side(ap.start,ap.end,ap.case,firstpoint,midpoint,lastpoint)
+                    if add_indicies is None:
+                        ii += 1
+                        firstpoint=midpoint
+                        continue
+
+                    if len(add_indicies) == 1:
+                        target = add_indicies[0]
+                        case_a = add_cases[0]
+                        case_b = add_cases[1]
+                        if self.blocked(target,ap.start,ap.end):
+                            ii += 1
+                            firstpoint=midpoint
+                            continue
+                        else:
+                            edge_a = find_edge(ap.start,target,case_a)
+                            edge_b = find_edge(target,ap.end,case_b)
+                            if self.previous_diagonals(edge_a,edge_b,firstpoint,lastpoint):
+                                ii += 1
+                                firstpoint=midpoint
+                                continue
+                            self.remove(ii)
+                            self.add(ii,[edge_a,edge_b])
+                            path_length += 1
+                            converged = False
+                            continue
+                        
+                # Updating crossing point
+                midpoint_prime = self.newton_smooth(ap.start,ap.end,ap.case,firstpoint,midpoint,lastpoint)
+                if type(midpoint_prime) == type(None) or np.isnan(midpoint_prime[0]) or np.isnan(midpoint_prime[1]):
+                    midpoint_prime = midpoint
+                    #raise Exception('Newton call failed to converge or recover')
```

Comment:
Remove commented exception if no longer used

Reply:
I have reintroduced the raise Exception and remove `midpoint_prime = prime` as this was work started for the slow glider routing and was never finished. As this has not been fully tested I have revered back to when the codebase was validated for the faster vehicle